cloudinit/sources/__init__.py

```diff
@@ -800,9 +801,11 @@ def instance_id_matches_system_uuid(instance_id, field='system-uuid'):
         return False
 
     dmi_value = util.read_dmi_data(field)
+    previous = dmi_value.lower()
+    current = instance_id.lower()
     if not dmi_value:
         return False
-    return instance_id.lower() == dmi_value.lower()
+    return current == previous or current == byte_swapped(current, previous)
```

Comment:
It seems like maybe we should make this byte_swapping change check by subclassing cloudinit/sources/DataSourceAzure.check_instance_id before passing that value byte-swapped value in instead of forcing all clouds to exercise this additional logic that will never apply to them

Reply:
Wouldn't this cover the code flow test

 https://github.com/canonical/cloud-init/blob/master/tests/unittests/test_datasource/test_azure.py#L1099

If not something similar using self.assertTrue(dsaz.check_instance_id())

Reply:
> Wouldn't this cover the code flow test
> 
> https://github.com/canonical/cloud-init/blob/master/tests/unittests/test_datasource/test_azure.py#L1099
> 
> If not something similar using self.assertTrue(dsaz.check_instance_id())

Correct , you probably want something that asserts against saz.check_instance_id() in the case where bytes are swapped.